rocketpy/simulation/flight.py

```diff
@@ -2702,31 +2732,58 @@ def u_dot_parachute(self, t, u, post_processing=False):
         # Get the mass of the rocket
         mp = self.rocket.dry_mass
 
-        # to = 1.2
-        # eta = 1
-        # Rdot = (6 * R * (1 - eta) / (1.2**6)) * (
-        #     (1 - eta) * t**5 + eta * (to**3) * (t**2)
-        # )
-        # Rdot = 0
-
-        # tf = 8 * nominal diameter / velocity at line stretch
-
-        # Calculate added mass
-        ma = (
-            self.parachute_added_mass_coefficient
-            * rho
-            * (2 / 3)
-            * np.pi
-            * self.parachute_radius**2
-            * self.parachute_height
-        )
-
         # Calculate freestream speed
         freestream_x = vx - wind_velocity_x
         freestream_y = vy - wind_velocity_y
         freestream_z = vz
         free_stream_speed = (freestream_x**2 + freestream_y**2 + freestream_z**2) ** 0.5
 
+        # Initialize parachute geometrical parameters
+        radius = self.parachute_radius
+        height = self.parachute_height
+        inflated_radius = (
+            (3 * self.parachute_volume * radius) / (4 * math.pi * height)
+        ) ** (1 / 3)
+        inflated_height = inflated_radius * height / radius
+
+        # Calculate the surface area of the parachute
+        if radius > height:
+            e = math.sqrt(1 - (inflated_height**2) / (inflated_radius**2))
+            surface_area = (
+                math.pi * inflated_radius**2 * (1 + (1 - e**2) / e * math.atanh(e))
+            )
+        else:
+            e = math.sqrt(1 - (inflated_radius**2) / (inflated_height**2))
+            surface_area = (
+                math.pi
+                * inflated_radius**2
+                * (1 + inflated_height / (e * inflated_radius) * math.asin(e))
+            )
+
+        # Calculate volume flow of air into parachute
+        volume_flow = (
+            rho
+            * freestream_z  # considering parachute as vertical
+            * (
+                (math.pi * inflated_radius**2)
+                - (self.parachute_porosity * surface_area)
+            )
+        )
+
+        # Getting time step
```

Comment:
do you really need to store step.t0? Why?

Should this really be a public attribute?

Reply:
private is better